### PR TITLE
Added jax_numpy_union1d

### LINKED ISSUE #14846: union1d

### PR DESCRIPTION
**Pull request by Dev-Khant**
Closes #14846

### CODE REVIEW

**Comment by Dev-Khant** [comment]
@YushaArif99 Can you kindly review the PR and let me know of any changes. Thanks.

**Comment by Dev-Khant** [comment]
Hey @YushaArif99 can you review the PR. Thanks

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py [hunk 1]
```diff
@@ -468,3 +468,44 @@ def test_jax_numpy_unique(fn_inputs, frontend, test_flags, fn_tree, on_device):
...
+
+
+@handle_frontend_test(
+    fn_tree="jax.numpy.union1d",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        num_arrays=2,
+        shared_dtype=True,
+        min_num_dims=1,
+        max_num_dims=2,
```
Comment (YushaArif99) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py:480]: why limit the input generation so severely? I think it's best if we test for extreme cases as that will help iron out any subtle corner cases that might go unnoticed otherwise. 

Is there a particular reason you've chosen these values for the `max_num_dims` as well as the `min_value` and `max_value` parameters?

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @Dev-Khant 

the tests for `test_union1d` seem to be failing in the CI tests [here](https://github.com/unifyai/ivy/actions/runs/4881201992/jobs/8709816304?pr=14852).  Could you look into this and re-request a review once all relevant tests are passing. Thanks! 🙂

**Comment by Dev-Khant** [comment]
Hi @YushaArif99 sorry for delay but I have made the changes please do review it and let me know of any improvements. Thanks

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @Dev-Khant 

Its weird that the tests are passing locally for you since they are failing deterministically in the CI. Are you testing for the `paddlepaddle` backend locally? Because the CI test is failing on the `paddlepaddle` backend with some error regarding incorrect inputs to `scatter_nd`. You can read more about the error log [here](https://github.com/unifyai/ivy/actions/runs/5010585327/jobs/8980750648?pr=14852). You can also try rerunning the failing test (shown in the CI) locally to make debugging more easier. Thanks! 🙂

**Comment by YushaArif99** [comment]
Hi @Dev-Khant 

Are you still working on this PR? If not, we can then close this PR 🙂
